hicommonwealth/commonwealth
# Remove dead code relating to long-deprecated legacy social account login.

## Link to Issue
Closes: #4105 

## Description of Changes
- Removes all references to SocialAccount throughout codebase -- refers to a long-deprecated feature.
- Adds a migration to remove all database references to SocialAccount and drop the table.

## Test Plan
Technically we can't test the removal of a deprecated feature, but the following should suffice:
- Log in with Magic using a social link / email and confirm it works.
- Log in with a wallet and confirm it works.
- Once logged in with wallet, go to notification settings page. Add an email to the box to receive email notifications. Follow the link in your email, then refresh or return to the notifications settings page and ensure it allows you to select "Never" or "Weekly" (i.e. that it successfully linked your email). This piece is necessary because it is the remaining usage of the _legacy email-based login_ which was closely interrelated with the deprecated social login whose code this PR removes.

## Deployment Plan
Post-deploy, can remove the following environment vars if present (we still use DISCORD_CLIENT_ID for discobot):
```
  DISCORD_CLIENT_SECRET
  DISCORD_OAUTH_CALLBACK
  DISCORD_OAUTH_SCOPES
  GITHUB_CLIENT_ID
  GITHUB_CLIENT_SECRET
  GITHUB_OAUTH_CALLBACK
```

## Other Considerations

We should follow-up by figuring out a way to remove the legacy email-based login as well.

**Note that migration will permanently delete old SocialAccount data, but it is all extremely stale anyway and not used for any purpose + is sensitive.**
